sql/2020/12_Mobile_Web/12_01.sql

```diff
@@ -0,0 +1,22 @@
+#standardSQL
+# 12_01: 4G, 3G and 2G distribution
+SELECT
+  _4G,
+  _3G,
+  _2G
+FROM (
+  SELECT
+    COUNT(0) OVER () AS n,
+    ROUND(_4GDensity * 100, 2) AS _4G,
+    ROUND(_3GDensity * 100, 2) AS _3G,
+    ROUND(_2GDensity * 100, 2) AS _2G,
+    ROW_NUMBER() OVER (ORDER BY _4GDensity DESC) AS row
+  FROM
+    `chrome-ux-report.materialized.metrics_summary`
```

Comment:
Should we use `device_summary` instead, which will allow you to segment by phone and tablet devices? I assume you're not interested in desktop data here.

Reply:
We just want to know how often a site is accessed via 2/3/4G, regardless of the device. Would `device_summary` be best to use for this?

Reply:
In that case `metrics_summary` SGTM.

I'm not sure about the aggregation method though, this seems like you're sampling the distribution of connection densities, but not actually calculating any summary stats. How about calculating the 10,25,50,75,90th percentiles of origins' ECT?

Reply:
ECT?

Reply:
https://developer.mozilla.org/en-US/docs/Glossary/Effective_connection_type

Reply:
The main q we want to answer here is: Given a random website, what does the distribution of 2G, 3G and 4G visitors look like?

And then, what does this distribution look like for the top 10k sites? Is 4G used more often or less often.


Thinking about it more now, this question might be a bit flawed though...